pkgs/applications/window-managers/ragnarwm/default.nix

```
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeText
+, fontconfig
+, libX11
+, libXft
+, libXcursor
+, libXcomposite
+, conf ? null
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ragnarwm";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "cococry";
+    repo = "Ragnar";
+    rev = finalAttrs.version;
+    hash = "sha256-SZWhmFNmS2oLdO9BnPzimoind1452v/EEQzadc5A+bI";
+  };
+
+  buildInputs = [
+    fontconfig
+    libX11
+    libXft
+    libXcursor
+    libXcomposite
+  ];
+
+  prePatch = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/applications
+
+    substituteInPlace Makefile \
+      --replace '/usr/bin' "$out/bin" \
+      --replace '/usr/share' "$out/share"
+  '';
+
+  postPatch =
+  let
+    configFile =
+      if lib.isDerivation conf || builtins.isPath conf
+      then conf else writeText "config.h" conf;
+  in
+  lib.optionalString (conf != null) "cp ${configFile} config.h";
+
+  makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
+
+  meta = with lib; {
+    homepage = "https://ragnar-website.vercel.app";
+    description = "Minimal, flexible & user-friendly X tiling window manager";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ sigmanificient ];
+    changelog = "https://github.com/cococry/Ragnar/releases/tag/${finalAttrs.version}";
+    platforms = lib.platforms.linux;
```

Already using `with lib` for this attrset.
```suggestion
    platforms = platforms.linux;
```